### ISSUE TITLE
Hyperopt Based Hyperparameter Tuner

### ISSUE THREAD

**Issue by rbharath** [deepchem/deepchem#2684]
We're currently using hyperopt in the MoleculeNet hyperparameter search code (https://github.com/deepchem/moleculenet/blob/master/examples/gnn.py#L136). This offers some nice capabilities that DeepChem's current hyperopt code doesn't support. We should add a Hyperopt based hyperparameter tuner to DeepChem

**Comment by rahul13ramesh** [comment]
I'd be interested in implementing this!

**Comment by rbharath** [comment]
@rahul13ramesh Join on of the developer calls! https://forum.deepchem.io/t/joining-the-deepchem-developer-calls/354. We can brainstorm a design with you and help guide you to take a crack at it

**Comment by shaipranesh2** [comment]
is this issue still there, if so I can take a crack in it!!

**Comment by rbharath** [comment]
@shaipranesh2 Yes, this issue is still open! I'd recommend proposing a design in an issue and getting feedback on a developer call before starting an implementation since this is a more complex project
